pkg/util/hash/hash_test.go

```diff
@@ -124,3 +128,104 @@ func TestHashOutput(t *testing.T) {
 		t.Errorf("hash value didn't matched for the same list")
 	}
 }
+
+func TestHashObject(t *testing.T) {
+	// nil objects hash the same
+	require.Equal(t, HashObject(nil), HashObject(nil))
+
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "name",
+			Labels: map[string]string{
+				"a": "b",
+				"c": "d",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "container",
+					Env: []corev1.EnvVar{
+						{
+							Name:  "var1",
+							Value: "value1",
+						},
+					},
+				},
+			},
+		},
+	}
+	samePod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "name",
+			Labels: map[string]string{
+				"a": "b",
+				"c": "d",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "container",
+					Env: []corev1.EnvVar{
+						{
+							Name:  "var1",
+							Value: "value1",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// hashes are consistent
+	hash := HashObject(pod)
+	// same object
+	require.Equal(t, hash, HashObject(pod))
+	// different object but same content
+	require.Equal(t, hash, HashObject(samePod))
+
+	// /!\ hashing an object and its pointer lead to different values
+	require.NotEqual(t, hash, HashObject(&pod))
```

Comment:
IMO it should see only the values it shouldn't bother about pointers/values...

Reply:
that is just the test to differentiate the behavior and help us to understand better how the hash is working here , so its good to have. however its a test